File: Roadmap/01 - OPERADORES Y ESTRUCTURAS DE CONTROL/python/Erik-Anibal.py
 # EJERCICIO:
 # - Crea ejemplos utilizando todos los tipos de operadores de tu lenguaje:
 #   Aritméticos, lógicos, de comparación, asignación, identidad, pertenencia, bits...
 #   (Ten en cuenta que cada lenguaje puede poseer unos diferentes)
 # - Utilizando las operaciones con operadores que tú quieras, crea ejemplos
 #   que representen todos los tipos de estructuras de control que existan
 #   en tu lenguaje:
 #   Condicionales, iterativas, excepciones...
 # - Debes hacer print por consola del resultado de todos los ejemplos.
 #
 # DIFICULTAD EXTRA (opcional):
 # Crea un programa que imprima por consola todos los números comprendidos
 # entre 10 y 55 (incluidos), pares, y que no son ni el 16 ni múltiplos de 3.
 #
 # Seguro que al revisar detenidamente las posibilidades has descubierto algo nuevo.
 #

# Números del 10 al 55, pares, sin 16 ni múltiplos de 3

for i in range(10,56):
    if i %2 == 0 and i != 16 and i % 3 !=0:
        print(i)


# Operadores aritméticos, lógicos, de comparación, de asignación, de identidad, de pertenencia y de bits
suma = 5 + 3       # 8
resta = 5 - 3      # 2
multiplicacion = 5 * 3 # 15
division = 5 / 3   # 1.666...
modulo = 5 % 3     # 2
incremento = suma + 1  # 9
decremento = resta - 1  # 1

and_logico = True and False  # False
or_logico = True or False    # True
not_logico = not True        # False

igual = 5 == 5          # True
diferente = 5 != 3      # True
mayor = 5 > 3           # True
menor = 5 < 3           # False
mayor_igual = 5 >= 5    # True
menor_igual = 5 <= 3    # False

x = 5
x += 3  # 8
x -= 2  # 6
x *= 2  # 12
x /= 2  # 6.0
x %= 4  # 2

a = [1, 2, 3]
b = a
print("a es igual que b? ")
print(a is b)  # True

arr = [1, 2, 3]
print("el 2 está en el arreglo arr? ")
print(2 in arr)  # True


and_bits = 5 & 1    # 1
or_bits = 5 | 1     # 5
xor_bits = 5 ^ 1    # 4
not_bits = ~5       # -6
desplazamiento_izq = 5 << 1  # 10
desplazamiento_der = 5 >> 1  # 2

# Estructuras de control
import random

edad = random.randint(10, 80)

if edad >= 18:
    print("Eres mayor de edad")
elif edad >= 13:
    print("Eres adolescente")
else:
    print("Eres menor de edad")

diaNumerico = random.randint(1,7)
diaConLetras = "un dia de la semana"

if diaNumerico == 1:
    diaConLetras = "Domingo"
    print(diaNumerico, diaConLetras)
elif diaNumerico == 2:
    diaConLetras = "Lunes"
    print(diaNumerico, diaConLetras)
else:
    print(diaNumerico, "no es domingo ni lunes")

for i in range(5):
    print("For",i)

contador = 0

while contador < 5:
    print("while",contador)
    contador += 1

try:
    x = y + 1  # y no está definido
except NameError as error:
    print("Ha ocurrido un error: " + str(error))

File: Roadmap/01 - OPERADORES Y ESTRUCTURAS DE CONTROL/javascript/Erik-Anibal.js
/*
 * EJERCICIO:
 * - Crea ejemplos utilizando todos los tipos de operadores de tu lenguaje:
 *   Aritméticos, lógicos, de comparación, asignación, identidad, pertenencia, bits...
 *   (Ten en cuenta que cada lenguaje puede poseer unos diferentes)
 * - Utilizando las operaciones con operadores que tú quieras, crea ejemplos
 *   que representen todos los tipos de estructuras de control que existan
 *   en tu lenguaje:
 *   Condicionales, iterativas, excepciones...
 * - Debes hacer print por consola del resultado de todos los ejemplos.
 *
 * DIFICULTAD EXTRA (opcional):
 * Crea un programa que imprima por consola todos los números comprendidos
 * entre 10 y 55 (incluidos), pares, y que no son ni el 16 ni múltiplos de 3.
 *
 * Seguro que al revisar detenidamente las posibilidades has descubierto algo nuevo.
 */

// Números entre 10 y 55, solo pares, sin 16 ni múltiplos de 3

for (let i = 10; i <= 55; i++) {
    if(i == 16 || i % 3 == 0 ){
        }else if (i % 2 == 0){
            console.log("Números entre 10 y 55, solo pares, sin 16 ni múltiplos de 3: " + i);
        }
}

// Operadores aritméticos, lógicos, de comparación, de asignación, de identidad, de pertenecia y de bits
let suma = 5 + 3;       // 8
let resta = 5 - 3;      // 2
let multiplicacion = 5 * 3; // 15
let division = 5 / 3;   // 1.666...
let modulo = 5 % 3;     // 2
let incremento = ++suma;  // 9
let decremento = --resta;  // 1

let and = true && false;  // false
let or = true || false;   // true
let not = !true;          // false

let igual = 5 == "5";         // true
let igualEstricta = 5 === "5"; // false
let diferente = 5 != "5";     // false
let diferenteEstricta = 5 !== "5"; // true
let mayor = 5 > 3;            // true
let menor = 5 < 3;            // false
let mayorOIgual = 5 >= 5;     // true
let menorOIgual = 5 <= 3;     // false

let x = 5;
x += 3;  // 8
x -= 2;  // 6
x *= 2;  // 12
x /= 2;  // 6
x %= 4;  // 2

let objeto1 = {};
let objeto2 = objeto1;
console.log("objeto1 es identico a objeto2? ");
console.log(objeto1 === objeto2); // true

let arr = [1, 2, 3];
console.log("El 2 está en el arreglo arr? ");
console.log(2 in arr);  // true

let andBits = 5 & 1;    // 1
let orBits = 5 | 1;     // 5
let xorBits = 5 ^ 1;    // 4
let notBits = ~5;       // -6
let desplazamientoIzq = 5 << 1; // 10
let desplazamientoDer = 5 >> 1; // 2


// Condicionales if-else, switch
let edad;
edad = Math.floor(Math.random() * (80 - 10 + 1)) + 10;

if (edad >= 18) {
    console.log("Tienes " + edad + " Eres mayor de edad");
} else if (edad >= 13) {
    console.log("Tienes " + edad + " Eres adolescente");
} else {
    console.log("Tienes " + edad + " Eres menor de edad");
}

let diaEnNumero;
let diaEnTexto;

diaEnNumero = Math.floor(Math.random() * (7)) + 1;

switch (diaEnNumero) {
    case 1:
        diaEnTexto = "Domingo";
        console.log(diaEnNumero + " Hoy es " + diaEnTexto);
        break;
    case 2:
        diaEnTexto = "Lunes";
        console.log(diaEnNumero + " Hoy es " + diaEnTexto);
    break;
    case 3:
        diaEnTexto = "Martes";
        console.log(diaEnNumero + " Hoy es " + diaEnTexto);
    break;
    case 4:
        diaEnTexto = "Miércoles";
        console.log(diaEnNumero + " Hoy es " + diaEnTexto);
    break;
    case 5:
        diaEnTexto = "Jueves";
        console.log(diaEnNumero + " Hoy es " + diaEnTexto);
    break;
    case 6:
        diaEnTexto = "Viernes";
        console.log(diaEnNumero + " Hoy es " + diaEnTexto);
    break;
    case 7:
        diaEnTexto = "Sábado";
        console.log(diaEnNumero + " Hoy es " + diaEnTexto);
    break;
    default:
        console.log(diaEnNumero + " Hoy no se que día es");
}

// Ciclos For, While, Do While

for (let i = 0; i < 5; i++) {
    console.log("For " + i);
}

let contador = 0;

while (contador < 5) {
    console.log("While " + contador);
    contador++;
}

contador = 0;

do {
    console.log("Do While " + contador);
    contador++;
} while (contador < 5);

try {
    let x = y + 1; // y no está definido
} catch (error) {
    console.log("Ha ocurrido un error: " + error.message);
}
